File: OrchidCore/src/test/java/com/eden/orchid/testhelpers/BaseOrchidTest.java
package com.eden.orchid.testhelpers;

import com.caseyjbrooks.clog.Clog;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;

public class BaseOrchidTest {

    @BeforeEach
    public void setUp() {
        disableLogging();
    }

    @AfterEach
    public void tearDown() {
        enableLogging();
    }

    protected void disableLogging() {
        Clog.getInstance().setMinPriority(Clog.Priority.FATAL);
    }

    protected void enableLogging() {
        Clog.getInstance().setMinPriority(Clog.Priority.VERBOSE);
    }

}

File: OrchidTest/src/main/kotlin/com/eden/orchid/testhelpers/BaseOrchidTest.kt
package com.eden.orchid.testhelpers

import clog.Clog
import clog.dsl.setMinPriority
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.BeforeEach

interface BaseOrchidTest {

    @BeforeEach
    @JvmDefault
    fun baseSetUp() {
        disableLogging()
    }

    @AfterEach
    @JvmDefault
    fun baseTearDown() {
        enableLogging()
    }

    @JvmDefault
    fun disableLogging() {
        Clog.setMinPriority(Clog.Priority.FATAL)
    }

    @JvmDefault
    fun enableLogging() {
        Clog.setMinPriority(Clog.Priority.VERBOSE)
    }

}
